From 1a7bb2e3a2749bd709ea4cf10b66b6f6d05aaf9d Mon Sep 17 00:00:00 2001
From: James Browning <jamesb.fe80@gmail.com>
Date: Sun, 10 Apr 2022 07:53:37 -0700
Subject: [PATCH] I: NTPsec build does not repect --notests

STR: (run the following)
- git clone https://gitlab.com/NTPsec/ntpsec
- cd ntpsec
- ./waf configure build --notests -p

AR: NTPsec runs tests despite having a parameter telling it not to
ER: NTPsec should not run tests when it has been told not to
---
 wscript | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/wscript b/wscript
index c1d765046..d8a20b82e 100644
--- a/wscript
+++ b/wscript
@@ -1131,6 +1131,7 @@ def build(ctx):
     ctx.manpage(8, "ntpclients/ntpsnmpd-man.adoc")
 
     # Skip running unit tests on a cross compile build
+    from waflib import Options
     if not ctx.env.ENABLE_CROSS:
         # Force re-running of tests.  Same as 'waf --alltests'
         if ctx.cmd == "check":
@@ -1139,6 +1140,8 @@ def build(ctx):
             # Print log if -v is supplied
             if verbose > 0:
                 ctx.add_post_fun(test_print_log)
+        elif Options.options.no_tests:
+            return
 
         # Test binaries
         ctx.add_post_fun(bin_test)
@@ -1152,7 +1155,6 @@ def build(ctx):
         ctx.add_post_fun(bin_test_summary)
     else:
         pprint("YELLOW", "Unit test runner skipped on a cross-compiled build.")
-        from waflib import Options
         Options.options.no_tests = True
 
     if ctx.cmd == "build":
-- 
2.32.0

